main.py

```diff
@@ -44,28 +46,71 @@ def __init__(self):
             exit('Check if the configuration is set right')
 
     def _run(self):
-        for submission in self._get_submissions():
-            parsed_submission = TitleParser.create_parsed_submission_from_submission(submission)
+        while True:
+            for submission in self._get_submissions():
+                """ No submissions, drop into parsing Modmail for a bit """
+                if submission is None:
+                    break
 
-            if ExcludedArtists.is_excluded(parsed_submission):
-                continue
+                parsed_submission = TitleParser.create_parsed_submission_from_submission(submission)
 
-            if SelfPromoDetector.is_self_promo(parsed_submission, submission):
-                reply = templates.self_promotion.format(submission.author.name)
-                submission.mod.remove(False, mod_note="Self promotion")
-                self._reply_with_sticky_post(submission, reply)
-                continue
+                if ExcludedArtists.is_excluded(parsed_submission):
+                    continue
 
-            exceeds_threshold = self._do_service_checks(parsed_submission, submission)
+                if SelfPromoDetector.is_self_promo(parsed_submission, submission):
+                    reply = templates.self_promotion.format(submission.author.name)
+                    submission.mod.remove(False, mod_note="Self promotion")
+                    self._reply_with_sticky_post(submission, reply)
+                    continue
 
-            try:
-                if exceeds_threshold is False and parsed_submission.success is True:
-                    # Yeey this post probably isn't breaking the rules 🌈
-                    self._reply_with_sticky_post(submission, self.description_provider.get_reply(parsed_submission))
-            except DescriptionException:
-                # Can't find recording on Musicbrainz, skipping
-                # traceback.print_exc(file=sys.stdout)
-                pass
+                exceeds_threshold = self._do_service_checks(parsed_submission, submission)
+
+                try:
+                    if exceeds_threshold is False and parsed_submission.success is True:
+                        # Yeey this post probably isn't breaking the rules 🌈
+                        self._reply_with_sticky_post(submission, self.description_provider.get_reply(parsed_submission))
+                except DescriptionException:
+                    # Can't find recording on Musicbrainz, skipping
+                    # traceback.print_exc(file=sys.stdout)
+                    pass
+
+            for message in self.reddit.subreddit(self.config.SUBREDDIT).mod.stream.modmail_conversations(
```

Comment:
This here is the core of the new function.

Reply:
I'm going to see if I can make this run the submission part and the mod mail part in separate threads tonight. This feels out of place looking at the rest of the code

Reply:
👍 I agree it felt a bit clunky but I had seen the author make a big deal out of the lib not being thread safe.  The suggestions were running a whole other application or this intertwining.  

With that said, I'm not sure anything we are doing is going to care about thread safety much.